glean-core/android/src/main/java/mozilla/telemetry/glean/Glean.kt

```diff
@@ -166,7 +166,13 @@ open class GleanInternalAPI internal constructor () {
 
             // If any pings were registered before initializing, do so now.
             // We're not clearing this queue in case Glean is reset by tests.
-            pingTypeQueue.forEach { registerPingType(it) }
+            synchronized(this@GleanInternalAPI) {
```

Comment:
Gee, I now realize that `@synchronized` is at the top of the `initialize` function but we don't hold it here.

Reply:
Exactly.  It means we're only synchronized for the small bit of things we do on the caller thread, not on the coroutine.